Tests/Methods/Mesh/test_get_solution.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+from numpy import array, zeros, abs as np_abs
+from unittest import TestCase
+
+from SciDataTool import DataTime, Data1D, DataLinspace, VectorField
+
+from pyleecan.Classes.SolutionData import SolutionData
+from pyleecan.Classes.SolutionMat import SolutionMat
+from pyleecan.Classes.SolutionVector import SolutionVector
+
+
+@pytest.mark.MeshSol
+@pytest.mark.METHODS
+# @pytest.mark.DEV
+class Test_get_soltution(TestCase):
+    """ Tests for get_solution method from Solution classes"""
+
+    def test_SolutionMat(self):
+        DELTA = 1e-10
+
+        solution = SolutionMat()
+        solution.field = array([[1, 2, 3], [2, 3, 4]])
+        solution.axis_name = ["time", "indice"]
+        solution.axis_size = [2, 3]
+
+        # result without explicit solution indices
+        sol1 = solution.get_solution(indice=[0, 1])
+        sol2 = solution.get_solution(indice=[0, 1, 2])
+        sol3 = solution.get_solution(indice=[0, 1, 2, 3])
+
+        expected = array([[1, 2], [2, 3]])
+        result = np_abs(expected - sol1.field).sum()
+        msg = "Wrong result: returned " + str(sol1) + ", expected: " + str(expected)
+        self.assertAlmostEqual(result, 0, msg=msg, delta=DELTA)
+
+        expected = array([[1, 2, 3], [2, 3, 4]])
+        result = np_abs(expected - sol2.field).sum()
+        msg = "Wrong result: returned " + str(sol2) + ", expected: " + str(expected)
+        self.assertAlmostEqual(result, 0, msg=msg, delta=DELTA)
+
+        result = np_abs(expected - sol3.field).sum()
+        msg = "Wrong result: returned " + str(sol3) + ", expected: " + str(expected)
+        self.assertAlmostEqual(result, 0, msg=msg, delta=DELTA)
+
+        # result with explicit solution indieces
+        solution.indice = [0, 1, 2]
```

Comment:
If I understand correctly the purpose, maybe we could change the second part of the test to work with different indice triplet (such as a random 999, 2000, 11857)

Reply:
I don't know if this is what you meant, but I set pseudo random indices. That also discovered a bug with my method. 
So, thanks!

Reply:
It was exactely what I meant. Great :-)